arangod/Graph/BaseOptions.cpp

```diff
@@ -658,6 +688,39 @@ void BaseOptions::parseShardIndependentFlags(arangodb::velocypack::Slice info) {
   setMaxProjections(VPackHelper::getNumericValue<size_t>(
       info, StaticStrings::MaxProjections,
       DocumentProducingNode::kMaxProjections));
-  _vertexProjections = Projections::fromVelocyPack(info, "vertexProjections");
-  _edgeProjections = Projections::fromVelocyPack(info, "edgeProjections");
-}
+
+  {
+    if (!getVertexProjections().empty()) {
+      resourceMonitor().decreaseMemoryUsage(
+          getVertexProjections().size() * sizeof(aql::Projections::Projection));
+    }
+
+    _vertexProjections = Projections::fromVelocyPack(info, "vertexProjections");
+    try {
+      arangodb::ResourceUsageScope vGuard(
+          resourceMonitor(),
+          getVertexProjections().size() * sizeof(aql::Projections::Projection));
+      vGuard.steal();  // now we are responsible for tracking the memory
```

Comment:
Instead of using a guard and then stealing from it, simply call `increaseMemoryUsage(...)`.

Reply:
Added in: 60044d169796e5dd8850ba2454d5f44463b1fc76